### PR TITLE
Reworks overdose into an actual proc call

### PR DESCRIPTION
**Pull request by ghost**
Adds an actual overdose proc, for adding overdose effects to chemicals.

Right now, the PoC chems are

 - Bicaridine. Rather than its overdose effect being hidden in the deep savannah of organcode, it's now moved to where it should be
 - Methylin, giving you a headache while helping monkeys get their doctorate
 - Rezadone, making your head spin

:cl:
 * rscadd: Adds overdose proc to chems

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -24,8 +24,9 @@
 	var/sport = 1 //High sport helps you show off on a treadmill. Multiplicative
 	var/custom_metabolism = REAGENTS_METABOLISM
 	var/custom_plant_metabolism = HYDRO_SPEED_MULTIPLIER
-	var/overdose = 0
-	var/overdose_dam = 1
+	var/overdose_am = 0
+	var/overdose_tick = 0
+	var/tick
```
Comment (ghost) [inline code/modules/reagents/Chemistry-Reagents.dm:29]: This already exists, it's just called "data". See chloral.

**Comment by kilozombie** [comment]
I like what you've done with Cheesy Gloop but currently the shitcode I used to make it stop your heart was located in DEFINES/reagents.dm, under cheartstopper. You might want to remove that.

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (ghost) [reply code/modules/reagents/Chemistry-Reagents.dm:29]: Data's a list and used for other things such as blood transfusion. there's nothing that just naturally ticks upwards unless specified (like in your example, where data is reset to being an integer)

Comment (ghost) [reply code/modules/reagents/Chemistry-Reagents.dm:29]: Hah, that's dumb

**Review by unid15 — APPROVED** [review]

**Pull request merged**